### PR TITLE
setup test environment

### PR DESCRIPTION
**Pull request by edouard-lopez**
**related:** #35.

* add doc
* create first test
* add [travis](https://travis-ci.org/rafaelrinaldi/pure) badge

### CODE REVIEW

**Comment by edouard-lopez** [comment]
@rafaelrinaldi I haven't enough privileges to set up Travis:

> You don't have sufficient rights to enable this repo on Travis.
> Please contact the admin to enable it or to receive admin rights yourself.

**Comment by rafaelrinaldi** [comment]
@edouard-lopez That's a bummer, you should. There's no other options for bumping up you as an admin or something as far as I can see. I have created the project on Travis CI now, so perhaps try it again?

**Comment by edouard-lopez** [comment]
@rafaelrinaldi guess it need to be an organisation give other user admin privileges. Did you setup the service in the [github settings](https://github.com/rafaelrinaldi/pure/settings/installations)?

**Comment by edouard-lopez** [comment]
@rafaelrinaldi @oranja I'm waiting your review to merge this PR.
@rafaelrinaldi could you try to fix the travis setting on [test-__pure_set_default](https://github.com/rafaelrinaldi/pure/tree/test-__pure_set_default) branch ?

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+dist: xenial
+sudo: required
+before_install:
+    - sudo add-apt-repository -y ppa:fish-shell/release-2
```
Comment (rafaelrinaldi) [inline .travis.yml]: Do we really need `sudo` here?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    - sudo apt-get update
+    - sudo apt-get -y install fish
+script:
+    - curl -Lo ~/.config/fish/functions/fisher.fish --create-dirs git.io/fisherman
+    - fish -c "fisher fishtape .; fishtape tests/*.test.fish"
```
Comment (rafaelrinaldi) [inline .travis.yml]: Super interesting, I didn't know this.

**Comment by rafaelrinaldi** [comment]
@edouard-lopez Happy new year!

Looks good to me, just added a minor comment but you can go ahead.

File: .travis.yml (hunk 1 above)
Comment (edouard-lopez) [reply .travis.yml]: I took the [config from `fishtape`](https://github.com/fisherman/fishtape/blob/master/.travis.yml)

**Comment by rafaelrinaldi** [comment]
@edouard-lopez 🚢

**Pull request merged**